launcher/ui/widgets/ThemeCustomizationWidget.cpp

```diff
@@ -0,0 +1,130 @@
+    ui->widgetThemeLabel->setVisible(features & ThemeFields::WIDGETS);
+    ui->backgroundCatComboBox->setVisible(features & ThemeFields::CAT);
+    ui->backgroundCatLabel->setVisible(features & ThemeFields::CAT);
+}
+
+void ThemeCustomizationWidget::applyIconTheme(int index) {
+    emit currentIconThemeChanged(index);
+
+    auto settings = APPLICATION->settings();
+    auto original = settings->get("IconTheme").toString();
+    // FIXME: make generic
+    settings->set("IconTheme", m_iconThemeOptions[index]);
+
+    if (original != settings->get("IconTheme")) {
+        APPLICATION->applyCurrentlySelectedTheme();
+    }
+}
+
+void ThemeCustomizationWidget::applyWidgetTheme(int index) {
+    emit currentWidgetThemeChanged(index);
+
+    auto settings = APPLICATION->settings();
+    auto originalAppTheme = settings->get("ApplicationTheme").toString();
+    auto newAppTheme = ui->widgetStyleComboBox->currentData().toString();
+    if (originalAppTheme != newAppTheme) {
+        settings->set("ApplicationTheme", newAppTheme);
+        APPLICATION->applyCurrentlySelectedTheme();
+    }
+}
+
+void ThemeCustomizationWidget::applyCatTheme(int index) {
+    emit currentCatChanged(index);
+
+    auto settings = APPLICATION->settings();
+    switch (index) {
+        case 0: // original cat
+            settings->set("BackgroundCat", "kitteh");
+            break;
+        case 1: // rory the cat
+            settings->set("BackgroundCat", "rory");
+            break;
+        case 2: // rory the cat flat edition
+            settings->set("BackgroundCat", "rory-flat");
+            break;
+    }
+}
+
+void ThemeCustomizationWidget::applySettings()
+{
+    applyIconTheme(ui->iconsComboBox->currentIndex());
+    applyWidgetTheme(ui->widgetStyleComboBox->currentIndex());
+    applyCatTheme(ui->backgroundCatComboBox->currentIndex());
+}
+void ThemeCustomizationWidget::loadSettings()
+{
+    auto settings = APPLICATION->settings();
+
+    // FIXME: make generic
+    auto theme = settings->get("IconTheme").toString();
+    ui->iconsComboBox->setCurrentIndex(m_iconThemeOptions.indexOf(theme));
+
+    {
+        auto currentTheme = settings->get("ApplicationTheme").toString();
+        auto themes = APPLICATION->getValidApplicationThemes();
+        int idx = 0;
+        for (auto& theme : themes) {
+            ui->widgetStyleComboBox->addItem(theme->name(), theme->id());
+            if (currentTheme == theme->id()) {
+                ui->widgetStyleComboBox->setCurrentIndex(idx);
+            }
+            idx++;
+        }
+    }
+
+    auto cat = settings->get("BackgroundCat").toString();
+    if (cat == "kitteh") {
+        ui->backgroundCatComboBox->setCurrentIndex(0);
+    } else if (cat == "rory") {
+        ui->backgroundCatComboBox->setCurrentIndex(1);
+    } else if (cat == "rory-flat") {
+        ui->backgroundCatComboBox->setCurrentIndex(2);
+    }
+}
+
+void ThemeCustomizationWidget::retranslate()
+{
+    ui->retranslateUi(this);
+}
```

Comment:
Add newline

Reply:
I am not sure whats up with the newline, but I added it in #696a4df67efdae9641a9fe39295370937d7d3eb7 (the 2nd commit)
![image](https://user-images.githubusercontent.com/31988415/209980122-41e03fd1-ce7e-4404-87e1-d61cbcb7101d.png)
I'm a bit confused... 
![image](https://user-images.githubusercontent.com/31988415/209980191-6725e512-879f-49f6-b672-a99230dd11b1.png)

Reply:
imma just mark this as resolved as there is a newline.